src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -381,24 +381,54 @@ object RuleUtils {
       filesAppended: Seq[Path]): LogicalPlan = {
     // Transform the location of LogicalRelation with appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(fsRelation: HadoopFsRelation, baseOutput, _, _) =>
+      case baseRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            baseOutput,
+            _,
+            _) =>
+        val options = extractBasePath(location.partitionSpec) match {
+          case Some(basePath) =>
+            // Set "basePath" so that partitioned columns are also included in the output schema.
+            Map("basePath" -> basePath)
+          case _ =>
+            Map[String, String]()
+        }
```

Comment:
ni: Can we utilize `map`, `getOrElse`, `orNull`, etc. of `Option` if applicable (as long as it doesn't hurt the readability)?
```scala
        val options = extractBasePath(location.partitionSpec)
          .map { basePath =>
            // Set "basePath" so that partitioned columns are also included in the output schema.
            Map("basePath" -> basePath)
          }
          .getOrElse(Map()) // I believe this may not need type info anymore (didn't check), although either one is fine.
```